target/bin/rspamd-dkim

```diff
@@ -186,8 +214,8 @@ function _create_keys() {
 
 function _check_permissions() {
   # shellcheck disable=SC2310
-  if ! __do_as_rspamd_user ls "${BASE_DIR}" >/dev/null; then
-    _log 'warn' "The Rspamd user ('_rspamd') seems to be unable to list files in the keys directory ('${BASE_DIR}') - Rspamd may experience permission errors later"
+  if ! __do_as_rspamd_user ls "${RSPAMD_DMS_DKIM_D}" >/dev/null; then
```

Comment:
Is it necessary to run as rspamd user? You use it for the `rspamadm dkim_keygen ...` at line 199 and other commands.

Is there an issue with using the root user to do this like opendkim, then correct permissions if necessary afterwards?

Reply:
Here, it is necessary because at this point, we test whether the Rspamd user can list files inside the directory. The actual `chown _rspamd:_rspamd ...` comes earlier :)